openlibrary/templates/account/view.html

```diff
@@ -40,10 +53,11 @@ <h1 class="details-title">$header_title</h1>
       $if page:
         $:macros.databarView(page)
 
-      $if mb.is_public and mb.key in mb.PUBLIC_KEYS:
-        <div>
-          <a class="cta-btn cta-btn--unstyled" onclick="prompt('Copy share link to clipboard:', window.location.protocol + '//' + window.location.hostname + '$(ctx.path)');"><img class="icon-link__image" src="/static/images/icons/share.svg" width="20"> Share</a>
-        </div>
+      $if mb.is_public and ctx.get('share_links') and mb.key in mb.PUBLIC_KEYS:
+        $ share_content = icon_link("share-modal-link", "/static/images/icons/share.svg", _("Share"), login_redirect=False, ga_data="ModalLinkClick|ShareIcon")
+        $ page_url = request.home
+        $ share_links = ctx.get('share_links')
+        $:macros.ShareModal(share_links, page_url, share_content, "share-modal-wrapper", show_embed=False)
```

Comment:
```suggestion
      $if mb.is_public and mb.key in mb.PUBLIC_KEYS:
        $set_share_links(url=request.canonical_url, title=header_title, view_context=ctx)
        $if ctx.get('share_links'):
          $ share_str = _('Share')
          $ share_content = '<a class="cta-btn cta-btn--unstyled share-modal-link icon-link" href="javascript:;" data-ol-link-track="MyBooksModalLinkClick|ShareIcon"><img class="icon-link__image" src="/static/images/icons/share.svg" width="20"> %s</a>' % share_str
          $ page_url = request.home
          $ share_links = ctx.get('share_links')
          $:macros.ShareModal(share_links, page_url, share_content, "share-modal-wrapper", show_embed=False)
```
A few changes here:
- Only call `set_share_links` if there's a possibility that the "Share" link will be rendered on the page.
- Use `MyBooksModalLinkClick|ShareIcon` as the `data-ol-link-track` value.  Elements with these attributes, when clicked, update our site analytics.  Using a different value here will help us determine if the click happened on a book page or a "My Books" page.
- Use the localized "Share" string, instead of a hard-coded English string.  If you're interested, you can learn more about this [here](https://github.com/internetarchive/openlibrary/wiki/Internationalization-Contributor's-Guide-(i18n)).

Reply:
Ya makes much more sense, making the changes as suggested